### PR TITLE
fix: skip commit id and sync for signing keys

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**
- prevent commit id increment for signing keys
**- How I did it**
- modified at_commit_log.dart --> commit method to return -1 for public:signing_publickey and signing_privatekey
- modified_isRegexMatches in commit_log_keystore.dart to remove signing private key
- added unit tests to commit_log_test.dart
**- How to verify it**
- run the persistence unit tests

### CODE REVIEW

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/at_commit_log.dart [hunk 1]
```diff
@@ -32,7 +33,9 @@ class AtCommitLog implements AtLogType {
     // So return -1.
     // The private: and privatekey: are not synced. so return -1.
     if (!key.startsWith('public:__') &&
-        key.startsWith(RegExp('private:|privatekey:|public:_'))) {
+        (key.startsWith(RegExp(
+                'private:|privatekey:|public:_|public:signing_publickey')) ||
+            key.contains(AT_SIGNING_PRIVATE_KEY))) {
```
Comment (gkc) [inline at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/at_commit_log.dart]: Use of `contains` seems potentially buggy. Why not an exact match?

**Review by gkc — DISMISSED** [review]

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/at_commit_log.dart (hunk 1 above)
Comment (murali-shris) [reply at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/at_commit_log.dart]: format of the key is <current_atsign>:signing_privatekey<current_atsign>

Comment (murali-shris) [reply at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/at_commit_log.dart]:
```
key.startsWith(RegExp('@(?<sharedWith>.*):signing_privatekey@(?<sharedBy>.*)'))
```
is this fine?

Comment (gkc) [reply at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/at_commit_log.dart]: LGTM

File: at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart [hunk 2]
```diff
@@ -341,8 +341,7 @@ class CommitLogKeyStore
     if ((RegExp(regex).hasMatch(atKey)) ||
         atKey.contains(AT_ENCRYPTION_SHARED_KEY) ||
```
Comment (VJag) [inline at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart]: Should we refactor this to:

match =  _isRegexMatches() || _isSpecialKey();

_isSpecialKey contains checking for special keys

Comment (murali-shris) [reply at_secondary/at_persistence_secondary_server/lib/src/log/commitlog/commit_log_keystore.dart]: yeah..sounds good.i will make the changes

**Review by VJag — APPROVED** [review]

**Pull request merged**